components/discourse/discourse.app.js

```diff
@@ -0,0 +1,192 @@
+const get = require("lodash.get");
+const sortBy = require("lodash.sortby");
+const axios = require("axios");
+const { nanoid } = require("nanoid");
+
+module.exports = {
+  type: "app",
+  app: "discourse",
+  propDefinitions: {
+    categories: {
+      type: "string[]",
+      label: "Categories",
+      description:
+        "The Discourse categories you want to watch for changes. **Leave blank to watch all categories**.",
+      async options() {
+        // Categories endpoint does not implement pagination
+        const categories = await this.listCategories();
+        if (!categories.length) {
+          return [];
+        }
+        const rawOptions = categories.map((c) => ({
+          label: c.name,
+          value: c.id,
+        }));
+        const options = sortBy(rawOptions, ["label"]);
+
+        return {
+          options,
+        };
+      },
+    },
+    eventTypes: {
```

Comment:
This prop and the two that follow aren't used yet, but I'm planning to create some components where they will be.